### PR TITLE
fix: Psl entity score

### PR DESCRIPTION
**Pull request by Prachig-Microsoft**
This pull request overhauls the way extraction quality scores (`entity_score` and `schema_score`) are derived, stored, and displayed throughout the content processing pipeline. The main improvement is a new, robust scoring logic that ensures all completed runs yield meaningful numeric scores: using probabilistic confidence when available, and falling back to a structural completeness ratio otherwise. Failed or empty extractions consistently yield `0.0`. The changes also update API and UI code to clarify score semantics and handle edge cases gracefully. Comprehensive unit tests are included for the new logic.

**Scoring logic improvements:**

* Refactored score calculation in `SaveHandler` to always emit numeric scores for completed runs: uses probabilistic confidence if available, otherwise falls back to a structural completeness ratio; failed/empty runs yield `0.0`. Added `_derive_aggregate_scores` and `_is_filled_value` helpers for robust computation. [[1]](diffhunk://#diff-9a1e392e5a67246a51b3a3d836dbf7075a9f6121629612db6a8f5fa62af29fbdL115-R122) [[2]](diffhunk://#diff-9a1e392e5a67246a51b3a3d836dbf7075a9f6121629612db6a8f5fa62af29fbdR236-R317)
* Updated score assignment in `find_process_result` to use the new aggregate scoring logic.
* Added comprehensive unit tests for the new scoring logic in `test_save_handler_scores.py`.

**API and model documentation:**

* Clarified the meaning and fallback behavior of `entity_score` and `schema_score` in API models (`claim_process.py` in both API and workflow repositories). [[1]](diffhunk://#diff-d5beb8f502a21258125f2188da17c963f4e073275ba0e50e26421d5b0a51f6a3L57-R61) [[2]](diffhunk://#diff-eefdec6e7894b3e193aa086f9b0289140d4aea9b6f1575919a1cd0cefa292904L81-R85)

**UI and workflow robustness:**

* Updated the web UI (`ProcessQueueGrid.tsx`) to handle `null` or `undefined` scores gracefully, always displaying `0` if the score is missing. [[1]](diffhunk://#diff-7a5d3a043adff4db35c234ca00c097170968d7e2cf648b13ed904658c6e9f925L376-R380) [[2]](diffhunk://#diff-7a5d3a043adff4db35c234ca00c097170968d7e2cf648b13ed904658c6e9f925L385-R393)
* Improved workflow step (`document_process_executor.py`) to robustly coerce score payloads to floats, defaulting to `0.0` on errors or missing values. [[1]](diffhunk://#diff-82daad19dfbe3a8169d7cd52c72889cfc40681b4533923e8acf74d1ddabc6f99L245-R250) [[2]](diffhunk://#diff-82daad19dfbe3a8169d7cd52c72889cfc40681b4533923e8acf74d1ddabc6f99R260-L267)
* Minor: Updated score field comments and test coverage for default behaviors. [[1]](diffhunk://#diff-16e3f770f628bd5a0c7aa9ad8e723d6a657fa2fa1c268edb653c2c64796c7700L20-R20) [[2]](diffhunk://#diff-6baa92d6c7175387ae49ed6a16d8521ada9ef68d34be7b4a3657257d3d97fe4cR45)## Purpose

* ...

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Golden Path Validation
- [x] I have tested the primary workflows (the "golden path") to ensure they function correctly without errors.

## Deployment Validation
- [x] I have validated the deployment process successfully and all services are running as expected with this change.

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: src/ContentProcessor/src/libs/utils/azure_credential_utils.py [hunk 1]
```diff
@@ -19,7 +19,6 @@
 from azure.identity import (
     AzureCliCredential,
     AzureDeveloperCliCredential,
-    DefaultAzureCredential,
     ManagedIdentityCredential,
 )
```
[the remark this answers is not in the document]
Comment (Shreyas-Microsoft) [reply src/ContentProcessor/src/libs/utils/azure_credential_utils.py:23]: Not needed

File: src/ContentProcessor/src/libs/utils/credential_util.py [hunk 2]
```diff
@@ -19,7 +19,6 @@
 from azure.identity import (
     AzureCliCredential,
     AzureDeveloperCliCredential,
-    DefaultAzureCredential,
     ManagedIdentityCredential,
 )
```
[the remark this answers is not in the document]
Comment (Shreyas-Microsoft) [reply src/ContentProcessor/src/libs/utils/credential_util.py:23]: Not needed

File: src/ContentProcessorWeb/src/Pages/DefaultPage/Components/ProcessQueueGrid/ProcessQueueGrid.tsx [hunk 3]
```diff
@@ -382,7 +386,11 @@ const ProcessQueueGrid: React.FC<GridComponentProps> = () => {
                                                         <CustomCellRender
                                                             type="percentage"
                                                             props={{
-                                                                valueText: doc.schema_score.toString(),
+                                                                valueText:
+                                                                    doc.schema_score === null ||
+                                                                    doc.schema_score === undefined
+                                                                        ? "0"
+                                                                        : doc.schema_score.toString(),
```
[the remark this answers is not in the document]
Comment (Shreyas-Microsoft) [reply src/ContentProcessorWeb/src/Pages/DefaultPage/Components/ProcessQueueGrid/ProcessQueueGrid.tsx:393]: Not needed

File: src/ContentProcessorWeb/src/Pages/DefaultPage/Components/ProcessQueueGrid/ProcessQueueGrid.tsx [hunk 4]
```diff
@@ -373,7 +373,11 @@ const ProcessQueueGrid: React.FC<GridComponentProps> = () => {
                                                         <CustomCellRender
                                                             type="percentage"
                                                             props={{
-                                                                valueText: doc.entity_score.toString(),
+                                                                valueText:
+                                                                    doc.entity_score === null ||
+                                                                    doc.entity_score === undefined
+                                                                        ? "0"
+                                                                        : doc.entity_score.toString(),
```
[the remark this answers is not in the document]
Comment (Shreyas-Microsoft) [reply src/ContentProcessorWeb/src/Pages/DefaultPage/Components/ProcessQueueGrid/ProcessQueueGrid.tsx:380]: Not needed

File: src/ContentProcessorWorkflow/src/utils/credential_util.py [hunk 5]
```diff
@@ -126,11 +125,11 @@ def get_azure_credential():
...
     )
-    
+
     raise RuntimeError(
-    "No Azure authentication available. "
-    "Use Managed Identity in Azure or run "
-    "'az login' / 'azd auth login' locally."
+        "No Azure authentication available. "
+        "Use Managed Identity in Azure or run "
+        "'az login' / 'azd auth login' locally."
```
[the remark this answers is not in the document]
Comment (Shreyas-Microsoft) [reply src/ContentProcessorWorkflow/src/utils/credential_util.py:132]: Not needed

File: src/ContentProcessor/src/libs/pipeline/handlers/save_handler.py [hunk 6]
```diff
@@ -241,3 +233,85 @@ def _summarize_processed_time(self, step_results: list[StepResult]) -> str:
...
+
+        # Path 2: structural completeness fallback
+        if comparison_items:
+            filled = sum(
+                1
+                for item in comparison_items
+                if SaveHandler._is_filled_value(item.Extracted)
+            )
+            ratio = round(filled / len(comparison_items), 3)
+            return (ratio, ratio, ratio)
```
[the remark this answers is not in the document]
Comment (Shreyas-Microsoft) [reply src/ContentProcessor/src/libs/pipeline/handlers/save_handler.py:314]: Not needed

File: src/ContentProcessor/src/libs/pipeline/handlers/save_handler.py [hunk 7]
```diff
@@ -241,3 +233,85 @@ def _summarize_processed_time(self, step_results: list[StepResult]) -> str:
...
+            else []
+        )
+
+        # Path 1: probabilistic confidence
+        if total_evaluated_fields_count > 0 and comparison_items:
+            zero_count = confidence.get("zero_confidence_fields_count", 0)
+            schema_score = round(
+                (len(comparison_items) - zero_count) / len(comparison_items),
+                3,
+            )
```
[the remark this answers is not in the document]
Comment (Shreyas-Microsoft) [reply src/ContentProcessor/src/libs/pipeline/handlers/save_handler.py:299]: Not needed

**Review by Avijit-Microsoft — APPROVED** [review]
